types/office-js/index.d.ts

```diff
@@ -28,126 +37,243 @@ declare namespace Office {
-         * Host is Access
+         * The Office host is Microsoft Access.
          */
         Access
     }
+    /**
+     * Specifies the OS or other platform on which the Office host application is running.
+     * 
+     * @remarks
+     * Hosts: Excel, Word, PowerPoint, Outlook, OneNote, Project, Access
+     */
     export enum PlatformType {
         /**
-         * Platform is PC
+         * The platform is PC (Windows).
          */
         PC,
         /**
-         * Platform is Web
+         * The platform is Office Online.
          */
         OfficeOnline,
         /**
-         * Platform is Mac
+         * The platform is Mac.
          */
         Mac,
         /**
-         * Platform is iOS
+         * The platform an iOS device.
          */
         iOS,
         /**
-         * Platform is Android
+         * The platform is an Android device.
          */
         Android,
         /**
-         * Platform is Winrt
+         * The platform is WinRT.
          */
         Universal
     }
     // Objects
-    export interface AsyncResult {
+        /**
+		* An object which encapsulates the result of an asynchronous request, including status and error information if the request failed.
+		*         
+        * @remarks
+        * Hosts: Access, Excel, Outlook, PowerPoint, Project, Word
+        *
+        * Last changed in: 1.1
+        * 
+        * When the function you pass to the `callback` parameter of an "Async" method executes, it receives an AsyncResult object that you can access from the `callback` function's only parameter.
+		*/
+        export interface AsyncResult {
+        /**
+        * Gets the user-defined item passed to the optional `asyncContext` parameter of the invoked method in the same state as it was passed in. This the user-defined item (which can be of any JavaScript type: String, Number, Boolean, Object, Array, Null, or Undefined) passed to the optional `asyncContext` parameter of the invoked method. Returns Undefined, if you didn't pass anything to the asyncContext parameter.
+        *         
+        * @remarks
+        * Hosts: Access, Excel, Outlook, PowerPoint, Project, Word
+        * 
+        * Last changed in: 1.1
+        */
         asyncContext: any;
+        /**
+        * Gets the [AsyncResultStatus](office.asyncresultstatus.md) of the asynchronous operation.
+        * 
+        * @remarks
+        * Hosts: Access, Excel, Outlook, PowerPoint, Project, Word
+        * 
+        * Last changed in: 1.1
+        */
         status: AsyncResultStatus;
+        /**
+        * Gets an [Error](office.error.md) object that provides a description of the error, if any error occurred.
+        * 
+        * @remarks
+        * Hosts: Access, Excel, Outlook, PowerPoint, Project, Word
+        * 
+        * Last changed in: 1.1
+        */
         error: Error;
+        /**
+        * Gets the payload or content of this asynchronous operation, if any.
+        
+        * @remarks
```

Comment:
Note:  the need to look up the resulting value would be completely eliminated when the d.ts is refactored to make use of Generics inside of AsyncResult (e.g., `AsyncResult<string>`, `AsyncResult<Office.Binding>`, etc.).
This is absolutely something that someone on the documentation team can do, it needn't be a dev.  I suggest you bring this up with Reeza [again].